sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -158,6 +158,7 @@ object Literal {
         Literal(CatalystTypeConverters.createToCatalystConverter(dataType)(v), dataType)
       case _: DayTimeIntervalType if v.isInstanceOf[Duration] =>
         Literal(CatalystTypeConverters.createToCatalystConverter(dataType)(v), dataType)
+      case _: ObjectType => Literal(v, dataType)
```

Comment:
shall we do some assertion here? e.g. `v.getClass` should be or a sub-class of `ObjectType.cls`

Reply:
> shall we do some assertion here? e.g. `v.getClass` should be or a sub-class of `ObjectType.cls`

Already have a check in `Literal.validateLiteralValue(value, dataType)` when constructing Literal